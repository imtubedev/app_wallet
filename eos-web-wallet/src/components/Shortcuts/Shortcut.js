import * as React from "react";
import { NavLink } from "react-router-dom";
import Shortcuts from "components/Shortcuts";
import css from "./styles.module.scss";

const activeClassName = css.active;

const Shortcut = (item, _key) => {
  if (Array.isArray(item)) {
    return (
      <li key={_key}>
        <Shortcuts className={css.logged_in} data={item} />
      </li>
    );
  } else {
    const { iconClass, text, key, ...props } = item;

    return (
      <li key={key}>
        <NavLink activeClassName={activeClassName} {...props}>
          <span className={iconClass} />
          {text}
        </NavLink>
      </li>
    );
  }
};

export default Shortcut;
