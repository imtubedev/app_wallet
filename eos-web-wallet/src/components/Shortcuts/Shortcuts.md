A list of shortcuts example:

```js
const { Provider } = require( "react-redux");
const { Router } = require("react-router-dom");
const { createBrowserHistory } = require("history");
const store = {
  getState: () => ({}),
  dispatch: _ => _,
  subscribe: _ => _
};
const noop = () => {};

<Provider store={store}>
  <Router history={createBrowserHistory()}>
    <div style={{ backgroundColor: "#575f68", padding: "2rem"}}>
      <Shortcuts data={[
        {
          to: "/",
          text: "Transfer",
          iconClass: "icon-transfer u-mr1"
        },
        {
          to: "/transactions",
          text: "Transaction History",
          iconClass: "icon-history u-mr1"
        },
        {
          to: "/accounts",
          text: "Accounts",
          iconClass: "icon-account u-mr1"
        }
      ]}
    />
    </div>
  </Router>
</Provider>
```
