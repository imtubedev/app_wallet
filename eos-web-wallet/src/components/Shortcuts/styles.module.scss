@import "../../styles/custom_variables.scss";

.menulist {
	li {
		position: relative;
		a {
			display: block;
			padding: 1.125rem 1.5rem;
			color: white;
			text-transform: uppercase;
			font-size: 0.875rem;
			font-weight: bold;
			transition: background .25s, padding .25s;
			&:after {
				position: absolute;
				bottom: 0;
				left: 0;
				content: "";
				height: 1px;
				width: 100%;
				background: linear-gradient(to right, $grey-60, $grey-80);
			}
			span {
				color: $primary;
				&:after {
					transition: border-left .25s;
					color: red;
					content: "";
					position: absolute;
					right: -8px;
					top: 0;
					width: 0;
					height: 0;
					border-top: 28px solid transparent;
					border-bottom: 28px solid transparent;
					border-left: 8px solid transparent;
					z-index: 1;
				}
			}
			&.active, &.active:hover {
				background-color: $orange;
				padding-left: 1.5rem;
				:before {
					color: white;
				}
				:after {
					border-left: 8px solid $orange;
					@include mobile {
						display: none;
					}
				}
			}
			&:hover {
				background-color: darken($grey-mid, 2%);
				padding-left: 2rem;
			}
		}
	}
	.menulist {
		a {
			text-transform: capitalize;
		}
	}
	&.logged_in {
		background-color: $grey-light;
		position: relative;
		box-shadow: none;
		&:before {
			position: absolute;
			top: 0;
			left: 0;
			content: "";
			height: 1px;
			width: 100%;
			background: linear-gradient(to right, $grey-40, $grey-60);
		}
		li {
			a {
				text-transform: capitalize;
				&:after {
					background: linear-gradient(to right, $grey-40, $grey-60);
				}
				&:hover {
					background-color: darken($grey-light, 2%);
				}
				&.active:hover {
					background-color: $orange;
				}
			}
		}
		&:after {
			position: absolute;
			width: 100%;
			height: 10px;
			content: "";
			box-shadow: inset 0px 10px 10px -5px rgba($grey-80,0.3);
		}
	}
}