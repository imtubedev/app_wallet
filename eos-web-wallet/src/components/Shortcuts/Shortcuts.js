import * as React from "react";
import List from "components/List";
import Shortcut from "./Shortcut";
import cx from "classnames";
import css from "./styles.module.scss";

const Shortcuts = ({
  className,
  data,
}) => (
  <List
    className={cx("menulist", css.menulist, className)}
    data={data}
    renderItem={Shortcut}
  />
);

export default Shortcuts;
