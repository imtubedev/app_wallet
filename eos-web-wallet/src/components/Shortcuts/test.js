import * as React from "react";
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import Shortcuts from './';

describe('<Shortcuts />', () => {
  it('renders without crashing', () => {
    const staticContext = {};
    const data = [
      // recursive item
      [
        {
          to: '/',
          text: 'Transfer',
          iconClass: 'icon-transfer u-mr1',
        },
      ],
      // stand alone shortuct
      {
        to: '/users',
        text: 'Users',
      },
    ];
    
    const tree = renderer.create(
      <StaticRouter location="/" context={staticContext}>
        <Shortcuts data={data} />
      </StaticRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})


