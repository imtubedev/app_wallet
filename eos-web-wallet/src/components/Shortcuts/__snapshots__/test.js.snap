// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shortcuts /> renders without crashing 1`] = `
<ul
  className="menulist"
>
  <li>
    <ul
      className="menulist"
    >
      <li>
        <a
          aria-current="true"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <span
            className="icon-transfer u-mr1"
          />
          Transfer
        </a>
      </li>
    </ul>
  </li>
  <li>
    <a
      aria-current={false}
      className={undefined}
      href="/users"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className={undefined}
      />
      Users
    </a>
  </li>
</ul>
`;
